ui/apps/platform/src/hooks/usePaginatedQuery.ts

```diff
@@ -0,0 +1,187 @@
+import { useState, useCallback, useEffect, useMemo } from 'react';
+import { debounce } from 'lodash';
+
+function filterNextPage<Item, ItemKey>(
+    results: Item[],
+    itemKeys: Set<ItemKey>,
+    keyFn: ((item: Item) => ItemKey) | undefined
```

Comment:
What do you think about `dedupKeyFn`
* Be consistent with function call.
* Let the name provide some of the meaning and let comment reinforce it.

Reply:
+1

I think this function was initially inside the hook and I used this naming to avoid shadowing.